Untangle core/css from babel

wereHamster opened the pull request:
Alternative fix for #599, much simpler than my previous attempt.

After this change, core/css.ts does not have any imports, it's a fully self-contained module. This very likely also improves typechecking performance for Linaria users as TypeScript won't have to load all the other modules (from Linaria or its dependencies, in particular babel).

TMaszko commented on src/core/css.ts:
```diff
@@ -1,4 +1,9 @@
+    className: string;
+    extends: StyledMeta;
+  };
+};
```
Or even duplicate it in both css and react :)

satya164 replied:
Wait, why does the `css` tag use this type

wereHamster replied:
I think it's to support things like this:

```
const Button = styled.button`
  color: red;
`;

const className = css`
& .${Button} {
  color: blue;
}
`;
```

satya164 replied:
Ok, makes sense. Let's just duplicate it

wereHamster commented:
Instead of duplicating it, I moved it to a separate file so that it can be imported where appropriate. I added a short jsdoc to the 'StyledMeta' type explaining where it is used. I also moved the `hasMeta` function into the same module, so it is closer to the type definition, as the two need to be kept in sync at all times.

I also found out that the type of StyledMeta was not accurate, [extends can be null](https://github.com/callstack/linaria/blob/e62668c3775bc930edb201d17bcf70940bd3a1f8/src/babel/evaluators/visitors/ProcessStyled.ts#L35-L37).
